x-pack/plugin/otel-data/build.gradle

```diff
@@ -4,9 +4,12 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-apply plugin: 'elasticsearch.internal-es-plugin'
-apply plugin: 'elasticsearch.internal-yaml-rest-test'
-apply plugin: 'elasticsearch.internal-cluster-test'
+plugins {
+  id 'elasticsearch.internal-es-plugin'
+  id 'elasticsearch.internal-yaml-rest-test'
+  id 'elasticsearch.internal-cluster-test'
+  id('com.google.protobuf') version '0.9.5'
```

Comment:
Bringing in an external plugin dependency is quite expensive in terms of maintenance. We rely on the fact that we can always use the latest gradle version without any deprecated or discouraged api usage. 

What does this plugin ultimately give us here? I'm inclined to inline the logic we need in build-tools-internal for better maintenance.

Reply:
@felixbarny lets keep the plugin for now. We can revisit later when we hit concrete issues. 

I changed a few things though to make this more gradle idiomatic, e.g. relying on gradle dependency management for downloading the otel proto zip and automatically unpacking via artifact transforms. that should give us more stability and cacheability than plain relying on hitting http endpoint each time. 

My change is at https://github.com/breskeby/elasticsearch/commit/fc77c0ec5878ff9bdfaea1e754338cc3e6abf09e

would you mind incorporating this?